src/managers/sessionManager.ts

```diff
@@ -0,0 +1,164 @@
+        regionId: sessionData.regionId || 3,
+        difficulty: sessionData.difficulty || 0,
+        hasStarted: sessionData.hasStarted || false,
+        enableVoice: sessionData.enableVoice || true,
+        matchType: sessionData.matchType || "NORMAL",
+        maps: sessionData.maps || [],
+        originalSessionId: sessionData.originalSessionId || "",
+        customSettings: sessionData.customSettings || "",
+        rewardSeed: sessionData.rewardSeed || -1,
+        guildId: sessionData.guildId || "",
+        buildId: sessionData.buildId || 4920386201513015989,
+        platform: sessionData.platform || 0,
+        xplatform: sessionData.xplatform || true,
+        freePublic: sessionData.freePublic || 3,
+        freePrivate: sessionData.freePrivate || 0,
+        fullReset: 0
+    };
+    sessions.push(newSession);
+    return newSession;
+}
+
+function getAllSessions(): Session[] {
+    return sessions;
+}
+
+function getSessionByID(sessionId: string): Session | undefined {
+    return sessions.find(session => session.sessionId === sessionId);
+}
+
+interface FindSessionRequest {
+    [key: string]: any;
+}
+
+function getSession(sessionIdOrRequest: string | FindSessionRequest): any[] {
+    if (typeof sessionIdOrRequest === "string") {
+        const session = sessions.find(session => session.sessionId === sessionIdOrRequest);
+        if (session) {
+            console.log("Found Sessions:", session);
+            return [
+                {
+                    createdBy: session.creatorId,
+                    id: session.sessionId
+                }
+            ];
+        }
+        return [];
+    } else {
+        const request = sessionIdOrRequest as FindSessionRequest;
+        const matchingSessions = sessions.filter(session => {
+            for (const key in request) {
+                if (key !== "eloRating" && key !== "queryId" && request[key] !== session[key as keyof Session]) {
+                    return false;
+                }
+            }
+            console.log("Found Matching Sessions:", matchingSessions);
+            return true;
+        });
+        return matchingSessions.map(session => ({
+            createdBy: session.creatorId,
+            id: session.sessionId
+        }));
+    }
+}
+
+function getSessionByCreatorID(creatorId: string): Session | undefined {
+    return sessions.find(session => session.creatorId === creatorId);
+}
+
+function getNewSessionID(): string {
+    const characters = "0123456789abcdef";
+    const maxAttempts = 100;
+    let sessionId = "";
+
+    for (let attempt = 0; attempt < maxAttempts; attempt++) {
+        sessionId = "64";
+        for (let i = 0; i < 22; i++) {
+            const randomIndex = Math.floor(Math.random() * characters.length);
+            sessionId += characters[randomIndex];
+        }
+
+        if (!sessions.some(session => session.sessionId === sessionId)) {
+            return sessionId;
+        }
+    }
```

Comment:
could remove this for loop and just use some type of UUID

Reply:
uuid package is really good, I can help to set it up.